api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -233,4 +238,29 @@ public PatientQueue pickPatientQueue(PatientQueue patientQueue, Provider provide
         patientQueue.setQueueRoom(queueRoom);
         return dao.savePatientQueue(patientQueue);
     }
+
+    /**
+     * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#
+     * getPatientQueueByParentLocation(org.openmrs.Location,
+     * org.openmrs.module.patientqueueing.model.PatientQueue.Status, java.util.Date dateFrom,
+     * java.util.Date)
+     */
+    @Override
+    public List<PatientQueue> getPatientQueueByParentLocation(Location parentLocation, PatientQueue.Status status, Date fromDate, Date toDate, boolean onlyInQueueRooms) {
+        LocationTag queueRomTag = Context.getLocationService().getLocationTagByUuid(ROOM_TAG_UUID);
+        Set<Location> childLocations = new HashSet<>();
+        if (onlyInQueueRooms) {
+            for (Location location : parentLocation.getChildLocations()) {
```

Comment:
What happens when getChildLocations returns null?

Reply:
Are you aware that this one includes retired child locations?

Reply:
Secondly, aren't you interested in child locations whose parents are other child locations?

Reply:
Aren't you also interested in child locations whose parents are other child locations?